build_system/src/config.rs

```diff
@@ -19,6 +23,72 @@ impl Channel {
     }
 }
 
+fn failed_config_parsing(err: &str) -> Result<ConfigFile, String> {
+    Err(format!(
+        "Failed to parse `{}`: {}",
+        ConfigFile::CONFIG_FILE,
+        err
+    ))
+}
+
+#[derive(Default)]
+pub struct ConfigFile {
+    gcc_path: Option<String>,
+    download_gccjit: Option<bool>,
+}
+
+impl ConfigFile {
+    pub const CONFIG_FILE: &'static str = "config.toml";
+
+    pub fn new() -> Result<Self, String> {
+        let content = fs::read_to_string(Self::CONFIG_FILE).map_err(|_| {
```

Comment:
This requires `config.toml` to be at the root of the cg_gcc source, right? When building as part of rustc that will not work. The source directory should be assumed to be immutable, thus preventing that file to be created by either the user or the rust build system. You could add a way to configure where it will look for `config.toml`.